src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -631,6 +635,14 @@ public virtual EntityTracker EntityTracker
         /// </summary>
         public virtual bool EnableWritingODataAnnotationWithoutPrefix { get; set; }
 
+        /// <summary>
+        /// Indicates whether a Where clause that just compares the id property generates a $filter query option.
+        /// </summary>
+        public virtual bool IdPredicateGeneratesFilterQueryOption

```

Comment:
Devs don't necessarily have any knowledge of what a "predicate" is wrt DSC - such vocabulary seems internal to the DSC implementation.
I'd consider renaming this to something like
`GenerateKeyCheckAsFilterQuery` or `GenerateKeyComparisonAsFilterQuery`

Reply:
I have switched to `KeyComparisonGeneratesFilterQuery`